t/oneliner.t

```diff
@@ -7,55 +7,117 @@ BEGIN {
+try_oneliner( q{my %h = (1, 2); print $h{1}},   [], q{2},   '%h and $h' );
 
 # switches?
-try_oneliner(q{print 'foo'}, ['-l'],           "foo\n",       'switches' );
+try_oneliner( q{print 'foo'}, ['-l'], "foo\n", 'switches' );
 
 # some DOS-specific things
-try_oneliner(q{print " \" "}, [],  q{ " },  'single quote' );
-try_oneliner(q{print " < \" "}, [],  q{ < " },  'bracket, then quote' );
-try_oneliner(q{print " \" < "}, [],  q{ " < },  'quote, then bracket' );
-try_oneliner(q{print " < \"\" < \" < \" < "}, [],  q{ < "" < " < " < },  'quotes and brackets mixed' );
-try_oneliner(q{print " < \" | \" < | \" < \" < "}, [],  q{ < " | " < | " < " < },  'brackets, pipes and quotes' );
+try_oneliner( q{print " \" "},   [], q{ " },   'single quote' );
+try_oneliner( q{print " < \" "}, [], q{ < " }, 'bracket, then quote' );
+try_oneliner( q{print " \" < "}, [], q{ " < }, 'quote, then bracket' );
+try_oneliner(
+    q{print " < \"\" < \" < \" < "}, [], q{ < "" < " < " < },
+    'quotes and brackets mixed'
+);
+try_oneliner(
+    q{print " < \" | \" < | \" < \" < "}, [],
+    q{ < " | " < | " < " < }, 'brackets, pipes and quotes'
+);
 
 # some examples from http://www.autohotkey.net/~deleyd/parameters/parameters.htm#CPP
-try_oneliner(q{print q[ &<>^|()@ ! ]}, [],  q{ &<>^|()@ ! },  'example 8.1' );
-try_oneliner(q{print q[ &<>^|@()!"&<>^|@()! ]}, [],  q{ &<>^|@()!"&<>^|@()! },  'example 8.2' );
-try_oneliner(q{print q[ "&<>^|@() !"&<>^|@() !" ]}, [],  q{ "&<>^|@() !"&<>^|@() !" },  'example 8.3' );
-try_oneliner(q{print q[ "C:\TEST A\" ]}, [],  q{ "C:\TEST A\" },  'example 8.4' );
-try_oneliner(q{print q[ "C:\TEST %&^ A\" ]}, [],  q{ "C:\TEST %&^ A\" },  'example 8.5' );
+try_oneliner( q{print q[ &<>^|()@ ! ]}, [], q{ &<>^|()@ ! }, 'example 8.1' );
+try_oneliner(
+    q{print q[ &<>^|@()!"&<>^|@()! ]}, [],
+    q{ &<>^|@()!"&<>^|@()! }, 'example 8.2'
+);
+try_oneliner(
+    q{print q[ "&<>^|@() !"&<>^|@() !" ]}, [],
+    q{ "&<>^|@() !"&<>^|@() !" }, 'example 8.3'
+);
+try_oneliner(
+    q{print q[ "C:\TEST A\" ]}, [], q{ "C:\TEST A\" },
+    'example 8.4'
+);
+try_oneliner(
+    q{print q[ "C:\TEST %&^ A\" ]}, [], q{ "C:\TEST %&^ A\" },
+    'example 8.5'
+);
 
 # XXX gotta rethink the newline test.  The Makefile does newline
 # escaping, then the shell.
 
+done_testing;
+exit;
+
+sub try_oneliner {
+    my ( $code, $switches, $expect, $name ) = @_;
+    my $cmd = $mm->oneliner( $code, $switches );
+    $cmd =~ s{\$\(ABSPERLRUN\)}{$perl};
+
+    # VMS likes to put newlines at the end of commands if there isn't
+    # one already.
+    $expect =~ s/([^\n])\z/$1\n/ if $^O eq 'VMS';
+
+    my $Makefile = makefile_name();
+
+    my $content = Makefile_template($cmd);
+    write_file( $Makefile, $content );
+
+    my $output = run(qq{$make -s all});
+
+    my $ok = $TB->is_eq( $output, $expect, $name )
+        || $TB->diag("$Makefile:\n$content");
+
+    return $ok;
+}
+
+sub Makefile_template {
+    my ($RUN) = @_;
+    my $NOECHO = '@';
```

Comment:
This will not be correct on all platforms

Reply:
Indeed you are right. This is kind a of a chicken egg issue: producing Makefile to test code which generate Makefile.. 
What would be your preference? 
- do we have some helpers we could use for this?
- skipping some distro for this test?
- going back to the unescape solution?
- other?

thanks for your valuable input

Reply:
Skipping and unescaping are not viable approaches. Probably the way to go here is to be making a sandbox dir, then minimal `Makefile.PL`. Look in the XS tests for functions to make this easy.